picard/tagger.py

```diff
@@ -517,76 +518,53 @@ def add_files(self, filenames, target=None):
             for file in new_files:
                 file.load(partial(self._file_loaded, target=target))
 
+    def _scan_dir(self, ignore_hidden, folders, recursive):
+        files = []
+        local_folders = list(folders)
+        while len(local_folders) > 0:
+            current_folder = local_folders.pop(0)
+            current_folder_files = []
+            for entry in os.scandir(current_folder):
+                if ignore_hidden and is_hidden(entry.path):
+                    continue
+                if entry.is_dir() and recursive:
```

Comment:
We might want to test for `recursive` first.

Reply:
Done.